core/async_job/async_job.ex

```diff
@@ -90,18 +92,50 @@ defmodule AntikytheraCore.AsyncJob do
       remaining_attempts: attempts,
       retry_interval:     Keyword.get(options, :retry_interval, RetryInterval.default()),
       payload:            payload,
+      immediate:          immediate_job?(options),
     ])
   end
 
+  defunp immediate_job?(options :: v[[option]]) :: boolean do
+    case options[:immediate] do
+      true ->
+        debug_assert(
+          [:schedule, :attempts, :retry_interval]
+          |> Enum.map(fn disallowed -> not Keyword.has_key?(options, disallowed) end)
+          |> Enum.all?(),
```

Comment:
`Enum.all?/2` can take transformer function so it can be written as:

```ex
Enum.all?([:schedule, :attempts, :retry_interval], fn disallowed -> not Keyword.has_key?(options, disallowed) end)
```

Reply:
Thank you, but we ended up to return errors instead of using `debug_assert`.

In 0a17fb1cc0e34322e8281898a5ec061cef1ecb01